homeassistant/components/hive/binary_sensor.py

```diff
@@ -34,62 +53,28 @@ async def async_setup_entry(
     devices = hive.session.deviceList.get("binary_sensor")
     entities = []
     if devices:
-        for dev in devices:
-            entities.append(HiveBinarySensorEntity(hive, dev))
+        for description in BINARY_SENSOR_TYPES:
+            for dev in devices:
+                if dev["hiveType"] == description.key:
+                    entities.append(HiveBinarySensorEntity(hive, dev, description))
     async_add_entities(entities, True)
 
 
 class HiveBinarySensorEntity(HiveEntity, BinarySensorEntity):
     """Representation of a Hive binary sensor."""
 
-    @property
-    def unique_id(self):
-        """Return unique ID of entity."""
-        return self._unique_id
-
-    @property
-    def device_info(self) -> DeviceInfo:
-        """Return device information."""
-        return DeviceInfo(
-            identifiers={(DOMAIN, self.device["device_id"])},
-            manufacturer=self.device["deviceData"]["manufacturer"],
-            model=self.device["deviceData"]["model"],
-            name=self.device["device_name"],
-            sw_version=self.device["deviceData"]["version"],
-            via_device=(DOMAIN, self.device["parentDevice"]),
-        )
-
-    @property
-    def device_class(self):
-        """Return the class of this sensor."""
-        return DEVICETYPE.get(self.device["hiveType"])
-
-    @property
-    def name(self):
-        """Return the name of the binary sensor."""
-        return self.device["haName"]
-
-    @property
-    def available(self):
-        """Return if the device is available."""
-        if self.device["hiveType"] != "Connectivity":
-            return self.device["deviceData"]["online"]
-        return True
-
-    @property
-    def extra_state_attributes(self):
-        """Show Device Attributes."""
-        return {
-            ATTR_MODE: self.attributes.get(ATTR_MODE),
-        }
-
-    @property
-    def is_on(self):
-        """Return true if the binary sensor is on."""
-        return self.device["status"]["state"]
+    def __init__(self, hive, hive_device, entity_description):
+        """Initialise hive binary sensor."""
+        super().__init__(hive, hive_device)
+        self.entity_description = entity_description
 
     async def async_update(self):
         """Update all Node data from Hive."""
         await self.hive.session.updateData(self.device)
         self.device = await self.hive.sensor.getSensor(self.device)
         self.attributes = self.device.get("attributes", {})
+        self._attr_is_on = self.device["status"]["state"]
+        if self.device["hiveType"] != "Connectivity":
```

Comment:
A future improvement, in another PR, is to add an attribute to the entity description that holds a function that accepts the device data and returns the availability boolean.